syftlib/src/main/java/org/openmined/syft/networking/clients/SocketClient.kt

```diff
@@ -0,0 +1,135 @@
+package org.openmined.syft.networking.clients
+
+import android.util.Log
+import io.reactivex.Single
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.processors.PublishProcessor
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonConfiguration
+import kotlinx.serialization.json.json
+import org.openmined.syft.processes.SyftJob
+import org.openmined.syft.networking.datamodels.NetworkModels
+import org.openmined.syft.networking.datamodels.SocketResponse
+import org.openmined.syft.networking.datamodels.webRTC.InternalMessageRequest
+import org.openmined.syft.networking.datamodels.webRTC.InternalMessageResponse
+import org.openmined.syft.networking.datamodels.webRTC.JoinRoomRequest
+import org.openmined.syft.networking.datamodels.webRTC.JoinRoomResponse
+import org.openmined.syft.networking.datamodels.syft.AuthenticationSuccess
+import org.openmined.syft.networking.datamodels.syft.CycleRequest
+import org.openmined.syft.networking.datamodels.syft.CycleResponseData
+import org.openmined.syft.networking.datamodels.syft.ReportRequest
+import org.openmined.syft.networking.datamodels.syft.ReportResponse
+import org.openmined.syft.networking.requests.MessageTypes
+import org.openmined.syft.networking.requests.Protocol
+import org.openmined.syft.networking.requests.REQUESTS
+import org.openmined.syft.networking.requests.SocketAPI
+import org.openmined.syft.networking.requests.WebRTCMessageTypes
+import org.openmined.syft.threading.ProcessSchedulers
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicBoolean
+
+private const val TAG = "SocketClient"
+
+@ExperimentalUnsignedTypes
+class SocketClient(
+    baseUrl: String,
+    private val timeout: UInt = 20000u,
+    private val schedulers: ProcessSchedulers
+) : SocketAPI {
+
+    //Choosing stable kotlin serialization over default
+    private val Json = Json(JsonConfiguration.Stable)
+
+    private val syftWebSocket = SyftWebSocket(Protocol.WSS, baseUrl, timeout)
+    @Volatile
+    private var socketClientSubscribed = AtomicBoolean(false)
+    private val messageProcessor = PublishProcessor.create<NetworkModels>()
+    private val compositeDisposable = CompositeDisposable()
+
+    private fun initiateSocketIfEmpty() {
```

Comment:
Better have the private functions down in the file to have a better view of the public API of the class.

Reply:
Is this function used anywhere?